mobly/controllers/android_device_lib/adb.py

```diff
@@ -41,6 +42,16 @@ def __str__(self):
                 ) % (self.cmd, self.ret_code, self.stdout, self.stderr)
 
 
+class AdbTimeoutError(AdbError):
```

Comment:
This should not inherit from AdbError because it does not provide its fields and also when someone writes `except AdbError` they don't mean to catch timeouts as well.

Reply:
Thanks for the catch.
Yes it should follow the same fields as `AdbError` if it inherits from `AdbError`.

I do think this should inherit from `AdbError` though as it was intended to signal an adb cmd has timed out.

Reply:
I found myself writing

```
except AdbTimeoutError:
  raise
except AdbError as e:
  use e.ret_code
```
And then thought "meh forget about the timeout".
From a practical perspective, this is worth considering, just to avoid this kind of workarounds. I think timeout is almost always fatal (catching it is unwanted) while a ret_code is not even always an error.

Reply:
PTAL #368